mycore-orcid2/src/main/java/org/mycore/orcid2/v3/rest/resources/MCRORCIDObjectResource.java

```diff
@@ -0,0 +1,226 @@
+import org.mycore.common.MCRSystemUserInformation;
+import org.mycore.datamodel.metadata.MCRMetadataManager;
+import org.mycore.datamodel.metadata.MCRObject;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.frontend.jersey.access.MCRRequireLogin;
+import org.mycore.frontend.jersey.filter.access.MCRRestrictedAccess;
+import org.mycore.orcid2.MCRORCIDUtils;
+import org.mycore.orcid2.client.MCRORCIDCredential;
+import org.mycore.orcid2.user.MCRORCIDSessionUtils;
+import org.mycore.orcid2.user.MCRORCIDUser;
+import org.mycore.orcid2.v3.MCRORCIDClientHelper;
+import org.mycore.orcid2.v3.MCRORCIDSectionImpl;
+import org.mycore.orcid2.v3.MCRORCIDWorkHelper;
+import org.mycore.orcid2.v3.MCRORCIDWorkSummaryUtils;
+import org.mycore.restapi.annotations.MCRRequireTransaction;
+import org.orcid.jaxb.model.v3.release.record.summary.Works;
+import org.orcid.jaxb.model.v3.release.record.summary.WorkSummary;
+
+/**
+ * Basic resource for general methods.
+ */
+@Path("/v1/")
+public class MCRORCIDObjectResource {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    /**
+     * Returns the publication status for a given MCRObjectID in the current user's ORCID profile, e.g.
+     *
+     * {
+     *   "isUsersPublication": true,
+     *   "isInORCIDProfile": true,
+     * }
+     *
+     * @param objectID the MCRObjectID
+     * @return the publication status
+     * @throws WebApplicationException if request fails
+     */
+    @GET
+    @Path("object-status/v3/{objectID}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @MCRRestrictedAccess(MCRRequireLogin.class)
+    public MCRORCIDPublicationStatus getPublicationStatus(@PathParam("objectID") MCRObjectID objectID) {
+        final MCRObject object = getObject(objectID);
+        final MCRORCIDUser orcidUser = MCRORCIDSessionUtils.getCurrentUser();
+        final Set<String> orcids = getMatchingORCIDs(object, orcidUser);
+        if (orcids.isEmpty()) {
+            // assumption that a publication cannot be in the profile, it is if not users.
+            return new MCRORCIDPublicationStatus(false, false);
+        }
+        if (orcids.size() > 1) { // should not happen
+            throw new WebApplicationException(Status.BAD_REQUEST);
+        }
+        final String orcid = orcids.iterator().next();
+        final MCRORCIDCredential credential = orcidUser.getCredentialByORCID(orcid);
+        if (credential == null) {
+            return new MCRORCIDPublicationStatus(true, false);
+        }
+        try {
+            final Works works = MCRORCIDClientHelper.getClientFactory().createUserClient(orcid, credential)
+                .fetch(MCRORCIDSectionImpl.WORKS, Works.class);
+            final List<WorkSummary> summaries
+                = works.getWorkGroup().stream().flatMap(g -> g.getWorkSummary().stream()).toList();
+            final boolean result = MCRORCIDWorkSummaryUtils.findMatchingSummariesByIdentifiers(object, summaries)
```

Comment:
@golsch Publications without an identifier won't be found that way. Maybe an additional check by putCode will do the trick.